Source/Core/UICommon/GameFile.h

```diff
@@ -52,8 +52,8 @@ class GameFile final
   const std::string& GetFilePath() const { return m_file_path; }
   const std::string& GetFileName() const { return m_file_name; }
   const std::string& GetName(const Core::TitleDatabase& title_database) const;
-  const std::string& GetName(bool long_name = true) const;
-  const std::string& GetMaker(bool long_maker = true) const;
+  const std::string& GetName(bool allow_custom_name, bool long_name = true) const;
+  const std::string& GetMaker(bool allow_custom_maker, bool long_maker = true) const;
```

Comment:
Those bools look iffy, especially since one of them is defaulted while the other is not. Maybe we should consider an enum instead; possibly even one that combines both as one (like `GetName(Long | AllowCustom)` or `GetName(Short)` etc.)

Thoughts?

Reply:
I could do enums, but we should avoid "combined" enums due to https://bugs.dolphin-emu.org/issues/11692.

Reply:
Does this also apply to `enum class`? I was just omitting stuff for brevity; I didn't necessarily mean for this to use an unqualified `enum`.

Reply:
With `enum class` I believe you get a compile error if you try to use `|` without casting, but I haven't tested it.

Reply:
Apparently you do get an error in that case, but you're allowed to declare your own `|` operator if you want it ([according to this article](http://blog.bitwigglers.org/using-enum-classes-as-type-safe-bitmasks/)).  That's probably excessive for something like this, though...

Reply:
We actually use this method in URDE with great success, we have a macro in our IO library athena which defines all of the necessary operators:
https://gitlab.axiodl.com/libAthena/athena/blob/master/include/athena/Global.hpp#L88-120
Then it's used like this:
https://gitlab.axiodl.com/AxioDL/hecl/blob/master/include/hecl/CVar.hpp#L27

Reply:
Both parameters are now using one `enum class`. (I thought merging them into one would fit, because the "custom and short" combination doesn't make sense since custom names don't have a limited length.)